packages/openstef-core/src/openstef_core/feature_engineering/validation_transforms/missing_values_transform.py

```diff
@@ -0,0 +1,209 @@
+from pydantic import ConfigDict, Field, model_validator
+from sklearn.impute import SimpleImputer
+
+from openstef_core.base_model import BaseConfig
+from openstef_core.datasets import TimeSeriesDataset
+from openstef_core.datasets.transforms import TimeSeriesTransform
+
+_logger = logging.getLogger(__name__)
+
+
+class ImputationStrategy(Enum):
+    """Enumeration of available imputation strategies for missing values."""
+
+    MEAN = "mean"
+    MEDIAN = "median"
+    MOST_FREQUENT = "most_frequent"
+    CONSTANT = "constant"
+
+
+class MissingValuesTransform(TimeSeriesTransform, BaseConfig):
+    """Transform that checks for, imputes and drops missing values in time series data.
+
+    This transform applies imputation strategies to handle missing values in the dataset.
+    For features specified in no_fill_future_values_features, it removes rows with
+    trailing NaN values instead of imputing them, as future values for these features
+    cannot be reasonably estimated.
+
+    Example:
+    >>> from datetime import timedelta
+    >>> import numpy as np
+    >>> import pandas as pd
+    >>> from openstef_core.datasets import TimeSeriesDataset
+    >>> from openstef_core.feature_engineering.validation_transforms.missing_values_transform import (
+    ...     ImputationStrategy,
+    ...     MissingValuesTransform,
+    ... )
+    >>> data = pd.DataFrame(
+    ...     {
+    ...         "radiation": [100, np.nan, 110, np.nan],
+    ...         "temperature": [20, np.nan, 24, 21],
+    ...         "wind_speed": [5, 6, np.nan, np.nan],
+    ...         "empty_feature": [np.nan, np.nan, np.nan, np.nan]
+    ...     },
+    ...     index=pd.date_range("2025-01-01", periods=4, freq="1h"),
+    ... )
+    >>> dataset = TimeSeriesDataset(data, timedelta(hours=1))
+    >>> transform = MissingValuesTransform(
+    ...     imputation_strategy=ImputationStrategy.MEAN,
+    ...     missing_value=np.nan,
+    ...     no_fill_future_values_features=["radiation"]
+    ... )
+    >>> transform.fit(dataset)
+    >>> result = transform.transform(dataset)
+    >>> int(result.data.isna().sum().sum()) == 0
+    True
+    >>> result.data.shape[0] == 3 # Last row dropped
+    True
+    >>> result.data.shape[1] == 3 # Empty feature column dropped
+    True
+    """
+
+    imputation_strategy: ImputationStrategy = Field(
+        ...,
+        description="The strategy to use for imputation",
+    )
+    missing_value: float = Field(
+        default=np.nan,
+        description="The placeholder for missing values that should be imputed",
+    )
+    fill_value: float | str | None = Field(
+        default=None,
+        description="Value to use when imputation_strategy is CONSTANT",
+    )
+    no_fill_future_values_features: list[str] = Field(
+        default_factory=list,
+        description="List of feature names for which trailing NaN values should not be filled",
+    )
+
+    imputer: SimpleImputer = Field(exclude=True, default_factory=SimpleImputer)
```

Comment:
This seems to be a private member var. Do you think it makes sense to prefix it as `_imputer`?